controllers/knative/knative.go

```diff
@@ -84,3 +102,100 @@ func GetKnativeAvailability(cfg *rest.Config) (*Availability, error) {
 		return result, nil
 	}
 }
+
+func GetWorkflowSink(ctx context.Context, c client.Client, workflow *operatorapi.SonataFlow, pl *operatorapi.SonataFlowPlatform) (*duckv1.Destination, error) {
+	if workflow == nil {
+		return nil, nil
+	}
+	if workflow.Spec.Sink != nil {
+		return workflow.Spec.Sink, nil
+	}
+	if pl != nil {
+		if pl.Spec.Eventing != nil {
+			// no sink defined in the workflow, use the platform broker
+			return pl.Spec.Eventing.Broker, nil
+		} else if pl.Status.ClusterPlatformRef != nil {
+			// Find the platform referred by the cluster platform
+			platform := &operatorapi.SonataFlowPlatform{}
+			if err := c.Get(ctx, types.NamespacedName{Namespace: pl.Status.ClusterPlatformRef.PlatformRef.Namespace, Name: pl.Status.ClusterPlatformRef.PlatformRef.Name}, platform); err != nil {
+				return nil, fmt.Errorf("error reading the platform referred by the cluster platform")
+			}
+			if platform.Spec.Eventing != nil {
+				return platform.Spec.Eventing.Broker, nil
+			}
+		}
+	}
+	return nil, nil
+}
+
+const knativeBrokerAnnotation = "eventing.knative.dev/broker.class"
+
+func GetKnativeResource(ctx context.Context, cfg *rest.Config, kRef *duckv1.KReference) (*unstructured.Unstructured, error) {
+	dynamicClient, err := dynamic.NewForConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+	gv, err := schema.ParseGroupVersion(kRef.APIVersion)
+	if err != nil {
+		return nil, err
+	}
+	resourceId := schema.GroupVersionResource{
+		Group:    gv.Group,
+		Version:  gv.Version,
+		Resource: kRef.Kind,
+	}
+	if len(kRef.Namespace) == 0 {
+		return nil, fmt.Errorf("namespace for knative resource %s is missing", kRef.Name)
+	}
+	list, err := dynamicClient.Resource(resourceId).Namespace(kRef.Namespace).List(ctx, metav1.ListOptions{})
+	fmt.Printf("list:%v", list)
+	return dynamicClient.Resource(resourceId).Namespace(kRef.Namespace).Get(ctx, kRef.Name, metav1.GetOptions{})
+}
+
+func IsKnativeBroker(kRef *duckv1.KReference) bool {
+	return kRef.APIVersion == "eventing.knative.dev/v1" && kRef.Kind == "Broker"
+}
+
+func IsKnativeEnvInjected(ctx context.Context, c client.Client, deploymentName, namespace string) (bool, error) {
```

Comment:
If this is a public function and the `Deployment` is ours, can you change the input param to receive a `SonataFlow` reference instead?

Reply:
You don't need to pass `Client` over, you can use this: https://github.com/apache/incubator-kie-kogito-serverless-operator/blob/main/utils/client.go#L25